control-plane/connect-inject/container_init.go

```diff
@@ -214,6 +219,19 @@ func (w *MeshWebhook) containerInit(namespace corev1.Namespace, pod corev1.Pod,
 		data.PrometheusScrapePath = prometheusScrapePath
 		data.PrometheusBackendPort = mergedMetricsPort
 	}
+	// Pull the TLS config from the relevant annotations.
```

Comment:
Are all of these annotations required or just one?
If they are all required, or required in sets we should enforce that with some validation and errors, in which case we should also update the unit tests accordingly.

Reply:
Cert, key, and one of CA file or path is required - would that validation go in the client/server daemonset templates, or somewhere else?

Reply:
Since it can be overridden by annotations I think we need it on both ends:
* in the template logic for the defaults
* here for the overrides

Reply:
I added the check/tests here - these won't be settable in the template/defaults though so this should be the only place the validation is needed.